### DISCUSSION TITLE
🙇‍♂️ Tasks and pipeline sections

### DISCUSSION THREAD

**Discussion by hejny** [webgptorg/promptbook#88] [Concepts]
[Pipeline](https://github.com/webgptorg/promptbook/discussions/64) is composed of multiple sections. Section can define a task or have non-task character.

There are 4 types of task sections, which are using [templating system](https://github.com/webgptorg/promptbook/discussions/17) (except Script).

- **LLM template** which call large language model
- **Simple template** which just concatenate strings
- **Script** which execute script like Javascript, Typescript or Python, [see more](https://github.com/webgptorg/promptbook/discussions/77)
- **Prompt dialog** which ask user for information, [see more](https://github.com/webgptorg/promptbook/discussions/76)
- *Non-task section* **Knowledge**, [see more](https://github.com/webgptorg/promptbook/discussions/41)
- *Non-task section*  **Sample parameter**, [see more](https://github.com/webgptorg/promptbook/discussions/43)

---

Workflow, scenario, chain or pipeline are synonyms, in Promptbook we prefer to use the word "Workflow"

**Comment by JorgeSquared** [comment]
Specify @hejny
